prow/github/client.go

```diff
@@ -107,6 +115,11 @@ type throttler struct {
 	lock     sync.RWMutex
 }
 
+// cacherTransport is needed for RoundTripper interface implemantation.
+type cacherTransport struct {
```

Comment:
The comment doesnt describe why we need a RoundTripper or what this one does. Also, what is this transport caching?